File: misc/rm_rf_exts.py
import os

def delete_files(directory, extensions):
    for root, dirs, files in os.walk(directory):
        for file in files:
            file_path = os.path.join(root, file)
            for extension in extensions:
                if file.endswith(extension.strip()):
                    print(f"Deleting file: {file_path}")
                    os.remove(file_path)
                    break

root_directory = input("Enter root directory [./]: ") or './'
file_extensions = input("Enter file extensions (comma-separated) [.save]: ") or '.save'
extensions = [extension.strip() for extension in file_extensions.split(',')]

print(f"\nPerforming recursive file deletion in '{root_directory}' with file extensions: {extensions}\n")

delete_files(root_directory, extensions)

File: misc/rm_rf_exts.js
const readline = require('readline');
const fs = require('fs');
const path = require('path');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question('Enter root directory [./]: ', (rootDirectory) => {
  rootDirectory = rootDirectory || './';

  rl.question('Enter file extensions (comma-separated) [.save]: ', (fileExtensions) => {
    fileExtensions = fileExtensions || '.save';
    const extensions = fileExtensions.split(',');

    console.log(`\nPerforming recursive file deletion in '${rootDirectory}' with file extensions: ${extensions}\n`);

    deleteFiles(rootDirectory, extensions);

    rl.close();
  });
});

function deleteFiles(directory, extensions) {
  fs.readdirSync(directory).forEach((file) => {
    const filePath = path.join(directory, file);
    const stats = fs.statSync(filePath);

    if (stats.isFile()) {
      for (const extension of extensions) {
        if (file.endsWith(extension.trim())) {
          console.log(`Deleting file: ${filePath}`);
          fs.unlinkSync(filePath);
          break;
        }
      }
    } else if (stats.isDirectory()) {
      deleteFiles(filePath, extensions);
    }
  });
}
